### ISSUE TITLE
Add mutli-domain support

### ISSUE THREAD

**Issue by angelodlfrtr** [NInfolab/alice#27]
Project should be able to handle multiple domains like this :

- `http://localhost:3000/domain1.tld/mypath`
- `http://localhost:3000/domain2.tld/mypath`
...

**Comment by nicoladiaz** [comment]
multiple domains = multiple sources streams through one server ?

**Comment by angelodlfrtr** [comment]
Yep

**Comment by angelodlfrtr** [comment]
Could be activated via config.

Two modes :

- Single domain, configured in config.json
- Multiple domain, handle any website

**Comment by nicoladiaz** [comment]
target or source  ?

**Comment by angelodlfrtr** [comment]
Primary target, but why not source

**Comment by angelodlfrtr** [comment]
And @NicolasPetton will update system, allowing dont set source in config. So source will not be required. Nothing will block multiple domains to point to server ip.

**Comment by nicoladiaz** [comment]
there is a question of terminology : alice streams the content of a domain "origin" (what we call target) to a new created domain (what we call source) hosted on a new server.

**Comment by angelodlfrtr** [comment]
Yep, idea is to allow multiple origin through http://created.domain/origin1.domain/, http://created.domain/origin2.domain/, etc.

**Comment by angelodlfrtr** [comment]
And http://created.domain1/origin1.domain/, http://created.domain2/origin1.domain/, http://created.domain1/origin2.domain/, http://created.domain2/origin1.domain/ will be possible

**Comment by angelodlfrtr** [comment]
With a single instance

**Comment by nicoladiaz** [comment]
with one powerfull instance :) so we need to stress test this kind of setup 'cause i doubt nodeJS will survive that way especially on a cheap VPS.

**Comment by angelodlfrtr** [comment]
Alice rewrite 'on the fly' packets, so not necessarily, it mostly depends on the traffic

**Comment by angelodlfrtr** [comment]
And in the setup script, we have to multiply server process according to number of cores of the machine, node is single threaded

**Comment by NicolasPetton** [comment]
angelodlfrtr <notifications@github.com> writes:

> Primary target, but why not source

What would be the use-case?

To support multiple targets, vhosts + multiple node instances would be a
IMHO a better way to go.

**Comment by angelodlfrtr** [comment]
In case where a domain is blocked by ISP

**Comment by angelodlfrtr** [comment]
No need to multiply instances

**Comment by nicoladiaz** [comment]
within the game "cats and mice" alice should remain agile and easy to migrate
